pkg/drain/drain.go

```diff
@@ -153,13 +155,27 @@ func (d *Helper) GetPodsForDeletion(nodeName string) (*podDeleteList, []error) {
 				// through any additional filters
 				break
 			}
+			if status.priority < priority {
+				//a previous filter had determined that this pod should be deleted at a lower priority
+				// use that priority
+				priority = status.priority
+			}
 		}
+		status.priority = priority
```

Comment:
The logic here is a little convoluted. It takes the `podDeleteStatus` of the last-called filter, except replaces the `priority` field with the lowest priority. Except if the `podDeleteStatus` indicates non-deletion, the priority of that last call is never considered, even using `podDeletionPriorityHighest` if the first filter indicated non-deletion.

I don't think the code should be modifying the returned `podDeleteStatus` structs.